### PR TITLE
Implementation of File.mkdir() and Fix for File.rm()

### PR DESCRIPTION
**Pull request by cookiengineer**
This implements File.mkdir(path) API and fixes the return value of File.rm();

Now both methods will return true on success and false on failure.
The test for File.mkdir() is also included, assuming that File/dir_to_create does not exist yet.

### CODE REVIEW

**Comment by cookiengineer** [comment]
Damn, gonna fix above issues first :)

File: src/Binding/JSFile.cpp [hunk 1]
```diff
@@ -287,9 +287,32 @@ bool JSFile::JS_isDir(JSContext *cx, JS::CallArgs &args)
...
+bool JSFile::JS_mkdir(JSContext *cx, JS::CallArgs &args)
+{
+
+    File *file = this->getFile();
+
+    int val = file->mkdir();
+    if (val == 0) {
+        args.rval().setBoolean(true);
+    } else {
+        args.rval().setBoolean(false);
```
Comment (paraboul) [inline src/Binding/JSFile.cpp]: `args.rval().setBoolean(val == 0);`

Comment (cookiengineer) [inline src/Binding/JSFile.cpp]: Jup, might be easier than the extra branch.

**Comment by verpeteren** [comment]
Please update the api-documentation


## For `File.rm`

Just add the return-type @ https://github.com/nidium/Nidium/blob/master/docs/JSFile.cpp.py#L122

```
ReturnDoc( "Returns `true` on success, `false` otherwise", "boolean" )
```

## For `File.mkDir`

Just add something like:

```
FunctionDoc( "File.mkDir", "Creates a directory",
    SeesDocs( "File.isDir|File.rmrf|File.listFiles" ),
    [ExampleDoc( """var result = File.mkDir('File/dir_to_create');
console.log("Directory created: ", result);
""", run_code=False ) ],
    IS_Static, IS_Public, IS_Fast,
    [ParamDoc("path", "File path", 'string', NO_Default, IS_Obligated)],
    ReturnDoc( "Returns `true` on success, `false` otherwise", "boolean" )
)
```

**Pull request closed without merging**